### PR TITLE
add common functions to test Addon config APIs

### LINKED ISSUE #2388: Create a library in tanzu-framework for addons API testing
Library should provide for the following capabilities:
1. Library to check if package installs are successful
2. Library to check if addon configs are good
3. Ability to customize the Addon Configs

### PR DESCRIPTION
**Pull request by ggpaue**
### Which issue(s) this PR fixes

Fixes #2388 

### Release note

```release-note
package-based-lcm: add common functions to test addon packages
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/test/tkgctl/shared/package.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+		return err
+	}
+
+	// Get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, cl, constants.TkgNamespace, mngCluster.Name)
+
+	// packageInstall name for for both management and workload clusters should follow the <cluster name>-<addon short name>
+	// packageInstall name and version should match info in clusterBootstrap for all packages, format is <package name>.<package version>
+	switch {
+	case testPkgName == "CNI":
```
Comment (shyaamsn) [inline pkg/v1/tkg/test/tkgctl/shared/package.go:95]: This function doesnt need to take in testPkgName as input. Instead can just get package name and versions from CB and verify if they are successful.

i.e.

```
cnipkgShortName, cnipkgName, cnipkgVersion, err = getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)

csipkgShortName, csipkgName, csipkgVersion, err = getPackageDetailsFromCBS(clusterBootstrap.Spec.CSI.RefName)

verifyPackageInstall(cnipkgShortName, cnipkgName, cnipkgVersion)
verifyPackageInstall(csipkgShortName, csipkgName, csipkgVersion)

func verifyPackageInstall() {

pkgiName := addonutil.GeneratePackageInstallName(mngCluster.Name, pkgShortName)
	pkgi := getPackageInstall(ctx, remoteClient, constants.TkgNamespace, pkgiName)

	// check package install reconcile status is succeed
	Expect(pkgi.Status.GenericStatus.Conditions[1].Type).Should(Equal(kappctrl.ReconcileSucceeded))
	Expect(pkgi.Status.GenericStatus.Conditions[1].Status).Should(Equal(corev1.ConditionTrue))

	// Verify package name match between clusterBootstrap and packageInstall
	Expect(pkgName).Should(Equal(pkgi.Spec.PackageRef.RefName))

	// Verify package version match between clusterBootstrap and packageInstall
	Expect(pkgVersion).Should(Equal(pkgi.Spec.PackageRef.VersionSelection.Constraints))
}
```

File: pkg/v1/tkg/test/tkgctl/shared/e2e_common.go [hunk 2]
```diff
@@ -150,6 +155,14 @@ func E2ECommonSpec(context context.Context, inputGetter func() E2ECommonSpecInpu
 		By(fmt.Sprintf("Waiting for workload cluster %q nodes to be up and running", clusterName))
 		framework.WaitForNodes(framework.NewClusterProxy(clusterName, tempFilePath, ""), 2)
 
+		By(fmt.Sprintf("Verify addon packages on workload cluster %q matches clusterBootstrap info on management cluster", clusterName))
+		scheme := runtime.NewScheme()
+		err = pkgiv1alpha1.AddToScheme(scheme)
+		Expect(err).NotTo(HaveOccurred())
+		// check antrea package information
+		err = checkPackageInstalls(context, client, scheme, "antrea")
```
Comment (shyaamsn) [inline pkg/v1/tkg/test/tkgctl/shared/e2e_common.go:163]: This e2e_common seems to be called for both legacy (non clusterclass) and clusterclass based clusters. For now lets only call this function if its clusterclass.i.e. if cluster plan == devcc. If not lets skip this. 

Later we need to enhance this function to work for legacy clusters as well

File: pkg/v1/tkg/test/tkgctl/shared/package.go (hunk 1 above)
Comment (ggpaue) [reply pkg/v1/tkg/test/tkgctl/shared/package.go]: modified with new structure and don't need to input package name, will check all packages at once

File: pkg/v1/tkg/test/tkgctl/shared/e2e_common.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggpaue) [reply pkg/v1/tkg/test/tkgctl/shared/e2e_common.go]: don't have to pass package name now, will check all packages at once

**Comment by ggpaue** [comment]
package-based-lcm branch has been merged into main branch, so open a PR towards main directly: 
https://github.com/vmware-tanzu/tanzu-framework/pull/2712

**Pull request closed without merging**